tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -109,6 +109,22 @@ def read(self, filename, binary_mode=False, size=None, offset=None):
             else:
                 return f.read()
 
+    def write(self, filename, file_content, binary_mode=False):
+        """Writes string file contents to a file.
+
+        Args:
+            filename: string, a path
+            file_content: string, the contents
+            binary_mode: bool, write as binary if True, otherwise text
+        """
+        mode = "wb" if binary_mode else "w"
+        if binary_mode:
```

Comment:
For a new API I'd prefer to be strict and only accept six.binary_type if binary_mode is True, otherwise only accept six.text_type.  Ditto for the S3 implementation.

Reply:
I'm a little bit nervous about this since we'll be passing down from torch.utils.tensorboard. We can do this in the GFile class implementation instead if you want.